### PR TITLE
[FLASK] Update Snaps privacy notice

### LINKED ISSUE #19500: Update privacy notice
Figma: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=198%3A20288&t=fvceI0kXSFIwM17t-1

- _Cancel button_ closes the entire connection attempt / window, and the same message will be displayed again on any future connection attempts (until it's accepted).
- _Accept button_ is disabled until the user scrolls to the bottom / or clicks the scroll-down-arrow (same logic as we use on install and update screens).

### PR DESCRIPTION
**Pull request by david0xd**
## Explanation
This PR will update Snaps privacy notice content and UX as requested. This notice is the one displayed on first snap installation or until user accepts the terms of use displayed in the popup.

Fixes https://github.com/MetaMask/metamask-extension/issues/19500

Figma designs: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=198-20452&t=3qMge4mvjrh6nAIS-0

Note: Scroll icon color is updated on the install screen as well to match the requirement and look the same.

### After
Screenshot 2023-06-08 at 12 48 16
Screenshot 2023-06-08 at 12 48 30
Screenshot 2023-06-08 at 12 49 31

### Before
Screenshot 2023-06-08 at 14 07 52
Screenshot 2023-06-08 at 14 08 06

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by Mrtenz — DISMISSED** [review]
Code looks good to me. I have not reviewed the design.

**Comment by eriknson** [comment]
Design lgtm!

**Comment by eriknson** [comment]
Design reviewed and ready to go.

**Review by Mrtenz — DISMISSED** [review]

File: ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js [hunk 1]
```diff
@@ -19,109 +19,129 @@ import {
...
-                {t('click')}
+          <Box className="snap-privacy-warning__message">
+            <Text variant={TextVariant.bodyMd}>
+              {t('snapsPrivacyWarningFirstMessage', [
                 <ButtonLink
+                  key="privacyNoticeTermsOfUseLink"
                   size={BUTTON_LINK_SIZES.INHERIT}
-                  onClick={handleReadMoreClick}
-                  data-testid="snapsPrivacyPopup_readMoreButton"
+                  href="https://consensys.net/terms-of-use/"
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-privacy-warning/snap-privacy-warning.js:82]: Can we move this to a variable stored in `shared/lib/ui-utils.js` ?

**Review by GuillaumeRx — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**